R/skip-vcr-off.R

```diff
@@ -0,0 +1,27 @@
+#' Skip tests if vcr is off
+#'
+#' Custom testthat skippers to skip tests if vcr is turned off via the
+#' environment variable `VCR_TURN_OFF`.
+#'
+#' @details This might be useful if your test will fail with real requests: when
+#'  the cassette was e.g. edited (a real request produced a 200 status code but
+#'  you made it a 502 status code for testing the behavior of your code when the
+#'   API errors)
+#'  or if the tests are very specific (e.g. testing
+#'  a date was correctly parsed, but making a real request would produce a different
+#'  date).
+#'
+#' @return Nothing, skip test.
+#' @export
+#'
+#' @seealso lightswitch
+skip_if_vcr_off <- function() {
+
+  if (is(try(find.package("testthat"), silent = TRUE), "try-error")) {
+    stop("This function is meant to be use within testthat tests. Install testthat.")
+  }
+
+  if (nzchar(Sys.getenv("VCR_TURN_OFF")) && as.logical(Sys.getenv("VCR_TURN_OFF"))) {
+    testthat::skip("Not run when vcr is off")
```

Comment:
if `testthat` is used within the package, and not just for tests, does `testthat` need to be in Imports?

Reply:
I do not think so since I check its installation and it is only needed for testthat users (you would not want to use this function if you did not already use testthat)